src/main/java/sg/ncl/MainController.java

```diff
@@ -1731,16 +1734,17 @@ public String checkJoinTeamInfo(
                 MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
                 ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
 
-                if (exceptionState == ExceptionState.TEAM_NOT_FOUND_EXCEPTION) {
-                    log.warn("join team request : team name error");
-                    redirectAttributes.addFlashAttribute(MESSAGE, "Team name does not exists.");
-                }
-                else if(exceptionState == ExceptionState.DETERLAB_OPERATION_FAILED_EXCEPTION) {
-                    log.warn("join team request: operation failed on DeterLab");
-                    redirectAttributes.addFlashAttribute(MESSAGE, "Error: " + (error.getMessage().contains("unknown error")? ERR_SERVER_OVERLOAD : error.getMessage()));
-                }
-                else {
-                    log.warn("join team request : some other failure");
+                if(checkUserException(exceptionState, error) != null) {
+                    log.info("Join team request : " + error.getMessage());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "Join team request : " 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)